src/dotnet/commands/dotnet-add/dotnet-add-p2p/Program.cs

```diff
@@ -13,114 +14,101 @@
 
 namespace Microsoft.DotNet.Tools.Add.ProjectToProjectReference
 {
-    public class AddProjectToProjectReferenceCommand
+    internal class AddProjectToProjectReferenceCommand : DotNetSubCommandBase
     {
-        internal static CommandLineApplication CreateApplication(CommandLineApplication parentApp)
-        {
-            CommandLineApplication app = parentApp.Command("p2p", throwOnUnexpectedArg: false);
-            app.FullName = LocalizableStrings.AppFullName;
-            app.Description = LocalizableStrings.AppDescription;
-            app.HandleRemainingArguments = true;
-            app.ArgumentSeparatorHelpText = LocalizableStrings.AppHelpText;
+        private CommandOption _frameworkOption;
 
-            app.HelpOption("-h|--help");
+        public static DotNetSubCommandBase Create()
+        {
+            var command = new AddProjectToProjectReferenceCommand()
+            {
+                Name = "p2p",
+                FullName = LocalizableStrings.AppFullName,
+                Description = LocalizableStrings.AppDescription,
+                HandleRemainingArguments = true,
+                ArgumentSeparatorHelpText = LocalizableStrings.AppHelpText,
+            };
 
-            CommandOption frameworkOption = app.Option(
-                $"-f|--framework <{CommonLocalizableStrings.CmdFramework}>",
-                LocalizableStrings.CmdFrameworkDescription,
-                CommandOptionType.SingleValue);
+            command.HelpOption("-h|--help");
 
-            app.OnExecute(() =>
-            {
-                try
-                {
-                    if (!parentApp.Arguments.Any())
-                    {
-                        throw new GracefulException(CommonLocalizableStrings.RequiredArgumentNotPassed, Constants.ProjectOrSolutionArgumentName);
-                    }
+            command._frameworkOption = command.Option(
+               $"-f|--framework <{CommonLocalizableStrings.CmdFramework}>",
+               LocalizableStrings.CmdFrameworkDescription,
+               CommandOptionType.SingleValue);
 
-                    var projectOrDirectory = parentApp.Arguments.First().Value;
-                    if (string.IsNullOrEmpty(projectOrDirectory))
-                    {
-                        projectOrDirectory = PathUtility.EnsureTrailingSlash(Directory.GetCurrentDirectory());
-                    }
+            return command;
+        }
 
-                    var projects = new ProjectCollection();
-                    var msbuildProj = MsbuildProject.FromFileOrDirectory(projects, projectOrDirectory);
+        public override int Run(string fileOrDirectory)
+        {
+            ProjectCollection projects = new ProjectCollection();
```

Comment:
I tend to prefer `var` because it makes refactoring easier and I get the sense that the coding convention is shifting in that direction but I'm new here. If you're of the `var`-sometimes opinion, this is still a pretty clear place where you're not losing information by using var, since the type being instantiated is specified right there. The class name is redundant in this example.